### PR TITLE
chore: Moves `bi_connector_config` and `advanced_configuration` in TPF to blocks

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Moves `bi_connector_config` and `advanced_configuration` in TPF to blocks

Link to any related issue(s): CLOUDP-290825

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals I have added appropriate changelog entries.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/service/advancedclustertpf/model_ClusterDescription20240805.go [hunk 1]
```diff
@@ -70,17 +70,26 @@ func NewTFModel(ctx context.Context, input *admin.ClusterDescription20240805, ti
 	}
 }
 
-func NewBiConnectorConfigObjType(ctx context.Context, input *admin.BiConnector, diags *diag.Diagnostics) types.Object {
+func restrictResourceModel(modelOut, modelIn *TFModel) {
```
Comment (EspenAlbert) [inline internal/service/advancedclustertpf/model_ClusterDescription20240805.go]: Can we use `overrideAttributesWithPrevStateValue` instead?
Which should probably be renamed to `overrideAttributesWithPlanValue`

File: internal/service/advancedclustertpf/model_to_ClusterDescriptionProcessArgsLegacy.go [hunk 2]
```diff
@@ -5,24 +5,25 @@ import (
...
-	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
 	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
 	admin20240530 "go.mongodb.org/atlas-sdk/v20240530005/admin"
 )
 
-func NewAtlasReqAdvancedConfigurationLegacy(ctx context.Context, objInput *types.Object, diags *diag.Diagnostics) *admin20240530.ClusterDescriptionProcessArgs {
+func NewAtlasReqAdvancedConfigurationLegacy(ctx context.Context, input *types.List, diags *diag.Diagnostics) *admin20240530.ClusterDescriptionProcessArgs {
 	var resp *admin20240530.ClusterDescriptionProcessArgs
-	if objInput == nil || objInput.IsUnknown() || objInput.IsNull() {
+	if input == nil || input.IsUnknown() || input.IsNull() || len(input.Elements()) == 0 {
```
Comment (EspenAlbert) [inline internal/service/advancedclustertpf/model_to_ClusterDescriptionProcessArgsLegacy.go]: tempting to create a helper: `listHasValues(input *types.List) bool` or `extractFirstValue[T](input *types.List) *T`

Comment (lantoli) [reply internal/service/advancedclustertpf/model_to_ClusterDescriptionProcessArgsLegacy.go]: I'm thinking of a bigger refactor in a follow-up PR to include more logic

Comment (lantoli) [reply internal/service/advancedclustertpf/model_to_ClusterDescriptionProcessArgsLegacy.go]: @EspenAlbert WDYT about this refactor? https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2909/commits/a829bf64112a58ec8e429d40d9115f1a47822648

Comment (EspenAlbert) [reply internal/service/advancedclustertpf/model_to_ClusterDescriptionProcessArgsLegacy.go]: Love it!

File: internal/service/advancedclustertpf/model_ClusterDescription20240805.go (hunk 1 above)
Comment (lantoli) [reply internal/service/advancedclustertpf/model_ClusterDescription20240805.go]: yes, that's better

File: internal/common/conversion/admin_generation.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package conversion
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func SingleListTFToSDK[TFModel, SDKRequest any](ctx context.Context, diags *diag.Diagnostics, input *types.List, fnTransform func(tf TFModel) SDKRequest) SDKRequest {
```
Comment (EspenAlbert) [inline internal/common/conversion/admin_generation.go:10]: Nice usage of Generic names!

**Review by EspenAlbert — APPROVED** [review]
Great! Did you do a manual check of the `terraform import` experience?

File: internal/common/conversion/admin_generation.go (hunk 3 above)
Comment (lantoli) [reply internal/common/conversion/admin_generation.go:10]: I followed your lead :-)

**Comment by lantoli** [comment]
> Great! Did you do a manual check of the terraform import experience?

yes, commented offline

**Pull request merged**